src/crypto/hashing/merkle.rs

```diff
@@ -110,12 +183,40 @@ impl Eq for MerkleTree {}
 mod test {
     use super::*;
     #[test]
-    fn test() {
+    fn merkle_tree_construction() {
         MerkleTree::new(&Vec::<u8>::new());
         MerkleTree::new::<u8>(&[1]);
         MerkleTree::new::<u8>(&[1, 2]);
         MerkleTree::new::<u8>(&[1, 2, 3]);
         MerkleTree::new::<u8>(&[1, 2, 3, 4]);
         MerkleTree::new::<u8>(&[1, 2, 3, 4, 5]);
     }
+
+    #[test]
+    fn child_size() {
+        assert_eq!(left_child_size(0), 0);
+        assert_eq!(left_child_size(1), 1);
```

Comment:
is left_child_size(1) really 1?

Reply:
I believe only a leaf node would have size of 1?

Reply:
You are correct. left_child_size(1) should be 0 (or it should panic for size < 2).